adobe/aepsdk-sample-app-android
# Implementation for Edge Identity ad ID + getUrlVariables features

## Description

This PR implements the new Edge Identity features for advertising identifier capabilities, and incorporates Android GAID functionality using the [Google Mobile Ads Lite SDK](https://developers.google.com/admob/android/lite-sdk).

Update Edge Identity view to add the following UI components: 
- Button `Update Ad ID With Current GAID` that initiates ad ID update flow
- Text that shows ad ID value (or error output)
- Text that shows ad ID tracking authorization state

Fetching device advertising identifier uses `ExecutorService` with a single thread executor and returns values using an `AdobeCallback`. This architecture was used since Google's API for getting ad ID is async and requires running on a background thread.

To avoid introducing a breaking change (crashing app on launch) to all users of the Android sample app, the implementation and SDK are commented out using special blocks; instructions on enabling the implementation and all prerequisites can be found in the [advertising identifier documentation](https://github.com/adobe/aepsdk-sample-app-android/blob/f9b6b69ef1f5b78c5bfdcbbf583639e67d30d6c6/Documentation/README.md).

It also implements the getUrlVariables feature (thanks to @cacheung and @addb) which returns the identifiers in URL query parameter format for consumption in hybrid mobile applications. The value returned from the API is shown under the button that implements this API. See the [documentation](https://github.com/adobe/aepsdk-edgeidentity-ios/blob/dev/Documentation/api-reference.md#geturlvariables) for more details on the API.

## Related Issue





Please see [MOB-16412](https://jira.corp.adobe.com/browse/MOB-16412) for additional details regarding the task.

## Motivation and Context

Sample app should be updated to showcase new Edge Identity advertising identifier features; both example of usage and enabling ad ID feature testing capabilities.

## How Has This Been Tested?




Tested functionality using Pixel_3a_API_32_arm64-v8a, using the button `Update Ad ID With Current GAID` to:
- Use Google Mobile Ads Lite SDK to get ad ID tracking authorization state
- Return the appropriate ad ID value and send through `MobileCore.setAdvertisingIdentifier`
- Verify results with expected values (sending the appropriate consent and ad ID events, responses, etc.)

## Screenshots (if appropriate):
From left to right:
1. Initial state
2. Ad tracking disabled
3. Ad tracking enabled
<p float="left">
<img width="250" alt="Initial state" src="https://user-images.githubusercontent.com/95260439/171560618-af718942-b934-479c-aa84-9f7a5a72182b.png">
<img width="250" alt="Ad tracking disabled" src="https://user-images.githubusercontent.com/95260439/171560634-423d378b-398f-4b72-a211-568610331fe2.png">
<img width="250" alt="Ad tracking enabled" src="https://user-images.githubusercontent.com/95260439/171560732-1a93b03f-49dc-4948-9356-69a1e0cddb4f.png">
</p>

## Types of changes




## Checklist:
